go/cmd/berty/mini/view_group.go

```diff
@@ -244,21 +258,32 @@ func (v *groupView) loop(ctx context.Context) {
 					continue
 				}
 
-				switch payload.GetType() {
-				case bertymessenger.AppMessageType_Acknowledge:
+				switch am.GetType() {
+				case bertymessenger.AppMessage_TypeAcknowledge:
 					if !bytes.Equal(evt.Headers.DevicePK, v.devicePK) {
 						continue
 					}
-
-					v.acks.Store(payload.(*bertymessenger.PayloadAcknowledge).Target, true)
+					var payload bertymessenger.AppMessage_Acknowledge
+					err := proto.Unmarshal(am.GetPayload(), &payload)
+					if err != nil {
+						v.logger.Error("failed to unmarshal Acknowledge", zap.Error(err))
+					}
+					v.acks.Store(payload.Target, true)
 					continue
 
-				case bertymessenger.AppMessageType_UserMessage:
-					receivedAt := time.Unix(0, payload.(*bertymessenger.PayloadUserMessage).SentDate*1000000)
+				case bertymessenger.AppMessage_TypeUserMessage:
+					var payload bertymessenger.AppMessage_UserMessage
+					err := proto.Unmarshal(am.GetPayload(), &payload)
+					if err != nil {
+						v.logger.Error("failed to unmarshal UserMessage", zap.Error(err))
+						continue
+					}
+
+					receivedAt := time.Unix(0, payload.SentDate*1000000)
```

Comment:
:+1:

Reply:
Finally, I prefer to stick with int64, because defining gorm models using protobuf is generic except for this part where we absolutely need gogoproto for now